clusters/build-clusters/01_cluster/cert-manager/apps-build01_certificate.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: cert-manager.io/v1alpha2
+kind: Certificate
+metadata:
+  labels:
+    aws-project: openshift-ci-infra
+  name: apps-build01-tls
+  namespace: openshift-ingress
+spec:
+  secretName: apps-build01-tls
+  issuerRef:
+    name: cert-issuer-aws
+    kind: ClusterIssuer
+  dnsNames:
+  - '*.apps.build01.ci.devcluster.openshift.com'
```

Comment:
Now we definitely need a controller otherwise all services get bricked once this expires, right?

Reply:
No.
This is the default DNS (*.apps.build01.ci.devcluster.openshift.com) setup.
In principle, its certs will be renewed automatically before they get expired. (This is the main task of cert-manager).

The case where we DO need the manager:
We do not like the default route's hostname, and we set up a new one by ingress (not route: ingress support secret-ref for certs while route does not). Then openshift-ingress controller will create route based on ingress and it supports only `termination: edge`.

So in short, if we have a route with customized hostname and `termination: reencrypt`, we need our own controller to implement secret-to-route logic.

```
oc get route -n ci release-controller release-controller-ocp release-controller-ocp-ppc64le release-controller-ocp-priv release-controller-ocp-s390x -o yaml | grep termination:
      termination: edge
      termination: edge
      termination: edge
      termination: reencrypt
      termination: edge
```

The bad boy is `release-controller-ocp-priv`. We do not have the custmized hostname for it.

https://github.com/openshift/release/blob/fb458601caab9976c340f6511e9c2ab15a2f17da/clusters/app.ci/cert-manager/release-controller_ingress.yaml#L10-L15

If we want to do the customized hostname for `release-controller-ocp-priv`, then we need our manager.

Reply:
But someone needs to copy the content of the secret into the route object once the certificate gets renewed or not?

Reply:
No need to copy anything when talking about the default hostnames.
That is the same as `acme.sh`: cert-manager refreshes the secret content and openshift components will notice the change on the secrets.

Reply:
I see, neat. Can we also set up the blackbox exporter to monitor these two or is that already the case?

Reply:
They are already included in the exporter:

https://github.com/openshift/release/blob/b6bfa8b2a12aa45281c5ab6aa3628e2b5441a01e/clusters/app.ci/prow-monitoring/additional-scrape-configs_secret.yaml#L14-L17

Let me remove the comment.

Reply:
Comments removed.

Reply:
@alvaroaleman 

Ready for the next round.